### PR TITLE
move dependency management to requirements.txt and requirements-dev.txt.

### PR DESCRIPTION
**Pull request by deathowl**
Add dockerfile and Docker related make targets

**What:**

- Move dependency management to requirements.txt and requirements-dev.txt to ease local development 
- Add related make targets: make install_build_deps and make install_deps
- Add Dockerfile 
- Add Docker related make targets to build, one-shot test built image, run container with bind-mounted sources in the background, enter said container:
Sample Docker run: https://gist.github.com/deathowl/e30bb72a791e362b3a8cf0e45e56c4f3
**Why:**

To ease development :) 
**How:**

**Risks:**
None
**Checklist**:

- [X] Added tests, if you've added code that should be tested
- [X] Updated the documentation, if you've changed APIs
- [X] Ensured the test suite passes
- [X] Made sure your code lints
- [X] Completed the Contributor License Agreement ("CLA")

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# set base image (host OS)
+FROM python:3.8
```
Comment (fornellas) [inline Dockerfile:2]: 3.9 (latest)?

**Comment by fornellas** [comment]
Absolutely 100% with having a container for development, awesome!

How do you see we maturing it? Right now requirements.txt are "as is", meaning we don't care about versions for most stuff. I thought of eventually updating the GH workflow to auto-update a `pip freeze` snapshot with all latest, so every push we do, we test with all latest versions (another PR).

**Review by fornellas — APPROVED** [review]

**Comment by fornellas** [comment]
PS: should we abandon #254 ? (I'm fine if you say so).

**Comment by deathowl** [comment]
@fornellas no, i think tox will be a huge added value, so i definitely wouldn't abandon it.

**Pull request merged**